storage/pebble/registers_pruner.go

```diff
@@ -0,0 +1,326 @@
+package pebble
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+)
+
+const (
+	DefaultPruneThreshold      = uint64(100_000)
+	DefaultPruneThrottleDelay  = 10 * time.Millisecond
+	DefaultPruneTickerInterval = 10 * time.Minute
+)
+
+// pruneIntervalRatio represents an additional percentage of pruneThreshold which is used to calculate pruneInterval
+const (
+	pruneIntervalRatio  = 0.1
+	deleteItemsPerBatch = 256
+)
+
+// pruneInterval is a helper function which calculates interval for pruner
+func pruneInterval(threshold uint64) uint64 {
+	return threshold + uint64(float64(threshold)*pruneIntervalRatio)
+}
+
+type RegisterPruner struct {
+	component.Component
+	componentManager *component.ComponentManager
```

Comment:
Doesn't look like we're using `componentManager` anywhere, so we don't need to store it
```suggestion
```

Reply:
[3ca7883](https://github.com/onflow/flow-go/pull/6397/commits/3ca788372b30910d6f1ea81fcf5e4566397a9634)